app/src/features/docs/components/DocsSearch.tsx

```diff
@@ -0,0 +1,396 @@
+      // Use the search API to get all matching packages from the registry
+      const results = await searchPackages(query);
+      
+      // Filter for packages that have documentation (docsIpfsUrl is not null)
+      const packagesWithDocs = results.filter(pkg => pkg.docsIpfsUrl !== null);
+      
+      // Sort results by relevance
+      const searchTerm = query.toLowerCase();
+      const sortedResults = packagesWithDocs.sort((a, b) => {
+        const aName = a.name.toLowerCase();
+        const bName = b.name.toLowerCase();
+        
+        // Exact matches first
+        if (aName === searchTerm && bName !== searchTerm) return -1;
+        if (bName === searchTerm && aName !== searchTerm) return 1;
+        
+        // Starts with search term
+        if (aName.startsWith(searchTerm) && !bName.startsWith(searchTerm)) return -1;
+        if (bName.startsWith(searchTerm) && !aName.startsWith(searchTerm)) return 1;
+        
+        // Contains search term
+        const aContains = aName.includes(searchTerm);
+        const bContains = bName.includes(searchTerm);
+        if (aContains && !bContains) return -1;
+        if (bContains && !aContains) return 1;
+        
+        // Alphabetical order as fallback
+        return aName.localeCompare(bName);
+      });
+
+      setSearchResults(sortedResults);
+    } catch (error) {
+      console.error('Search failed:', error);
+      setError('Failed to search packages');
+      setSearchResults([]);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  const handleSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchQuery(event.target.value);
+  };
+
+  // Helper function to calculate time since published
+  const getTimeAgo = (dateString: string) => {
+    const now = new Date();
+    const published = new Date(dateString);
+    const diffInMinutes = Math.floor((now.getTime() - published.getTime()) / (1000 * 60));
+    
+    // Handle future dates by treating them as "just now"
+    if (diffInMinutes <= 0) {
+      return 'just now';
+    }
+    
+    if (diffInMinutes < 60) {
+      return `${diffInMinutes} minute${diffInMinutes !== 1 ? 's' : ''} ago`;
+    } else if (diffInMinutes < 1440) {
+      const hours = Math.floor(diffInMinutes / 60);
+      return `${hours} hour${hours !== 1 ? 's' : ''} ago`;
+    } else {
+      const days = Math.floor(diffInMinutes / 1440);
+      return `${days} day${days !== 1 ? 's' : ''} ago`;
+    }
+  };
+
+  const renderSearchResults = (packages: PackageSearchResult[]) => (
+    <Box mt={4}>
+      {packages.map((pkg) => (
+        <Box
+          key={`${pkg.name}-${pkg.version}`}
+          sx={{
+            py: 2,
+            borderBottom: '1px solid',
+            borderColor: 'divider',
+            '&:last-child': { borderBottom: 'none' }
+          }}
+        >
+          <Box display="flex" alignItems="center" gap={1} mb={0.5}>
+            <Link
+              href={`/docs/${pkg.name}/${pkg.version}`}
```

Comment:
Current `Link`'s seem to have both `href` and an `onClick` (with `router.push`), which can potentially trigger double navigation or a full page reload.

Instead, can use `import NextLink from "next/link";` to link instead of router (could potentially remove the router import)
```suggestion
              component={NextLink}
              href={`/docs/${pkg.name}/${pkg.version}`}
```